app/models/katello/concerns/host_managed_extensions.rb

```diff
@@ -208,11 +208,13 @@ def sync_package_associations(new_installed_package_ids)
       end
 
       def import_tracer_profile(tracer_profile)
-        self.host_traces.destroy_all
+        host_traces.delete_all
+        traces = []
         tracer_profile.each do |trace, attributes|
-          self.host_traces.create!(:application => trace, :helper => attributes[:helper], :app_type => attributes[:type])
+          traces << { host_id: self.id, application: trace, helper: attributes[:helper], app_type: attributes[:type] }
         end
-        self.update_trace_status
+        Katello::HostTracer.import(traces)
```

Comment:
if we don't need validation (which we likely don't), you can add :validate => false to make it even faster.

Reply:
Good call, I don't think we need validation here